src/decimal-field.tsx

```diff
@@ -17,6 +17,25 @@ const getLocalDecimalChar = (locale: Locale): string => {
     }
 };
 
+const format = (value: any) => {
+    const sanitized = String(value ?? '').replace(/[^.,\d]/g, ''); // remove non digits or decimal separator chars;
+    const firstSeparator = /[.,]/.exec(sanitized);
+    const parts = sanitized.split(/[.,]/);
```

Comment:
do not make the localDecimalChar  replacement here, instead, keep the one the user typed. Make that replacement in `replace` prost-processor